smack-integration-test/src/main/java/org/jivesoftware/smackx/muc/AbstractMultiUserChatIntegrationTest.java

```diff
@@ -207,29 +211,42 @@ static void createNonAnonymousMuc(MultiUserChat muc, Resourcepart resourceName)
      * semi-anonymous (a value of "moderators"), or fully anonymous (a value of "none", not shown here).
      * </blockquote>
      */
-    static void createSemiAnonymousMuc(MultiUserChat muc, Resourcepart resourceName) throws SmackException.NoResponseException, XMPPException.XMPPErrorException, InterruptedException, MultiUserChatException.MucAlreadyJoinedException, SmackException.NotConnectedException, MultiUserChatException.MissingMucCreationAcknowledgeException, MultiUserChatException.NotAMucServiceException {
+    static void createSemiAnonymousMuc(MultiUserChat muc, Resourcepart resourceName) throws SmackException.NoResponseException, XMPPException.XMPPErrorException, InterruptedException, MultiUserChatException.MucAlreadyJoinedException, SmackException.NotConnectedException, MultiUserChatException.MissingMucCreationAcknowledgeException, MultiUserChatException.NotAMucServiceException, TestNotPossibleException {
         muc.create(resourceName);
         Form configForm = muc.getConfigurationForm();
         FillableForm answerForm = configForm.getFillableForm();
+        if (!answerForm.hasField("muc#roomconfig_whois")) {
+            throw new TestNotPossibleException(new MultiUserChatException.MucConfigurationNotSupportedException("muc#roomconfig_whois"));
+        }
         answerForm.setAnswer("muc#roomconfig_whois", "moderators");
         muc.sendConfigurationForm(answerForm);
     }
 
     /**
      * Creates a password-protected room.
      */
-    static void createPasswordProtectedMuc(MultiUserChat muc, Resourcepart resourceName, String password) throws SmackException.NoResponseException, XMPPException.XMPPErrorException, InterruptedException, MultiUserChatException.MucAlreadyJoinedException, SmackException.NotConnectedException, MultiUserChatException.MissingMucCreationAcknowledgeException, MultiUserChatException.NotAMucServiceException {
+    static void createPasswordProtectedMuc(MultiUserChat muc, Resourcepart resourceName, String password) throws SmackException.NoResponseException, XMPPException.XMPPErrorException, InterruptedException, MultiUserChatException.MucAlreadyJoinedException, SmackException.NotConnectedException, MultiUserChatException.MissingMucCreationAcknowledgeException, MultiUserChatException.NotAMucServiceException, TestNotPossibleException {
         muc.create(resourceName);
         Form configForm = muc.getConfigurationForm();
         FillableForm answerForm = configForm.getFillableForm();
-        answerForm.setAnswer("muc#roomconfig_passwordprotectedroom", true);
+
+        if (answerForm.hasField("muc#roomconfig_passwordprotectedroom")) {
+            // Prosody doesn't seem to use this field. Setting a room secret is likely 'enough'.
```

Comment:
After reading xep45, this seems like a bug in prosody to me.

Until its fixed, it should be guarded by a newly introduced 'prosody' `CompatilibytMode` (akin the two existing ones `standardsCompliant` and `ejabberd`.